code/game/objects/items/weapons/storage/bluespace.dm

```diff
@@ -0,0 +1,84 @@
+/*
+ *		Bag of Holding
+ *		Miniature black hole
+*/
+
+/obj/item/weapon/storage/backpack/holding
+	name = "Bag of Holding"
+	desc = "A backpack that opens into a localized pocket of Blue Space."
+	origin_tech = "bluespace=4"
+	item_state = "holdingpack"
+	icon_state = "holdingpack"
+	fits_max_w_class = 4
+	max_combined_w_class = 28
+
+/obj/item/weapon/storage/backpack/holding/miniblackhole
+	name = "miniature black hole"
+	desc = "A miniature black hole that opens into a localized pocket of Blue Space."
+	icon_state = "porthole"
+	slot_flags = 0
+	w_class = 2 //fits in pockets!
+
+/obj/item/weapon/storage/backpack/holding/suicide_act(mob/user)
+	user.visible_message("<span class = 'danger'><b>[user] puts \the [src.name] on \his head and stretches the bag around \himself. With a sudden snapping sound, the bag shrinks to its original size, leaving no trace of [user].</b></span>")
+	forceMove(get_turf(user))
```

Comment:
Yes it will, you should use an unequip proc.
I can't remember if suicide_act requires you to hold the item, because if so you can use user.drop_item(src)